src/main/java/hudson/plugins/ec2/win/EC2WindowsLauncher.java

```diff
@@ -73,8 +73,10 @@ protected void launchScript(EC2Computer computer, TaskListener listener) throws
             logger.println("remoting.jar sent remotely. Bootstrapping it");
 
             final String jvmopts = node.jvmopts;
-            final WindowsProcess process = connection.execute("java " + (jvmopts != null ? jvmopts : "") + " -jar "
-                    + tmpDir + AGENT_JAR, 86400);
+            final String remoteFS = node.getRemoteFS();
+            final String launchString = "java " + (jvmopts != null ? jvmopts : "") + " -jar " + tmpDir + AGENT_JAR + " -workDir \"" + remoteFS + "\"";
```

Comment:
You can use [WindowsUtil](https://github.com/jenkinsci/jenkins/blob/master/core/src/test/java/hudson/os/WindowsUtil.java) to properly quote arguments for both Windows and `cmd.exe` arguments (i.e., for use with `cmd.exe /c COMMAND` like with how you execute a builtin like `mklink`).

Reply:
I'd love to use it but unfortunately because its in the `/test` path of core its unusable here. Unless i fork that bit or have it moved into core. But even then core needs to be bumped again to use it.

Reply:
Oh crap, I forgot it was only a test class. Feel free to copy/paste the class with a TODO comment of moving it to jenkins-core at some point.

I wrote the initial utility class while rewriting some flaky IO-related tests as well as one regarding junction points in Windows (fairly similar to symbolic links but with Windows-specific filesystem features), but I bet we can find more improper quoting of Windows command line arguments throughout Jenkins that could benefit from it.